src/Atmos/Model/moisture.jl

```diff
@@ -45,89 +57,44 @@ Assumes the moisture components are computed via thermodynamic equilibrium.
 struct EquilMoist <: MoistureModel
 end
 vars_state(::EquilMoist,T) = NamedTuple{(:ρq_tot,),Tuple{T}}
-vars_gradient(::EquilMoist,T) = NamedTuple{(:q_vap, :q_liq, :q_ice, :temperature),Tuple{T,T,T,T}}
+vars_gradient(::EquilMoist,T) = NamedTuple{(:q_tot, :temperature),Tuple{T,T,T,T}}
 vars_diffusive(::EquilMoist,T) = NamedTuple{(:ρd_q_tot, :ρJ_ρD), Tuple{SVector{3,T},SVector{3,T}}}
 vars_aux(::EquilMoist,T) = NamedTuple{(:e_int, :temperature),Tuple{T,T}}
 
+get_phase_partition(::EquilMoist, state::Vars) = PhasePartition(state.ρq_tot/state.ρ)
+thermo_state(::EquilMoist, state::Vars, aux::Vars) = PhaseEquil(aux.moisture.e_int, state.ρq_tot/state.ρ, state.ρ, aux.moisture.temperature)
 
-function pressure(::EquilMoist, state::Vars, aux::Vars, t::Real)
-  ρ = state.ρ
-  q_tot = state.ρq_tot / ρ
-  phase = PhaseEquil(aux.moisture.e_int, q_tot, ρ, aux.moisture.temperature)
-  air_pressure(phase)
-end
-function soundspeed(::EquilMoist, state::Vars, aux::Vars, t::Real)
-  ρ = state.ρ
-  q_tot = state.ρq_tot / ρ
-  phase = PhaseEquil(aux.moisture.e_int, q_tot, ρ, aux.moisture.temperature)
-  soundspeed_air(phase)
-end
+function gradvariables!(m::EquilMoist, transform::Vars, state::Vars, aux::Vars, t::Real)
+  invρ = state.ρ
+  transform.moisture.q_tot = state.ρq_tot * invρ
 
-function gradvariables!(::EquilMoist, transform::Vars, state::Vars, aux::Vars, t::Real)
-  ρ = state.ρ
-  q_tot = state.ρq_tot / ρ
-  phase = PhaseEquil(aux.moisture.e_int, q_tot, ρ, aux.moisture.temperature)
+  phase = thermo_state(m, state, aux)
   q = PhasePartition(phase)
+  R_m = gas_constant_air(q)
+  T = aux.moisture.temperature
+  e_tot = state.ρe * invρ
 
-  transform.moisture.q_vap = q.tot - q.liq - q.ice
-  transform.moisture.q_liq = q.liq
-  transform.moisture.q_ice = q.ice
-  transform.moisture.temperature = aux.moisture.temperature
+  # a better name?
+  transform.moisture.e_x = e_tot + R_m*T
```

Comment:
@tapios is this what you had in mind?

Reply:
Yes, just a better name. `e_x` without the kinetic energy we call moist static energy. So maybe this should be `moist dynamical energy`, or `dyn_e_tot` or something like it.

Reply:
Or, perhaps better: "total enthalpy" (a slight abuse of terminology, because total enthalpy does not usually contain potential energy). It's strange this quantity does not have an established name...